plugin.video.embycon/resources/lib/context_monitor.py

```diff
@@ -0,0 +1,56 @@
+import threading
+import sys
+import xbmc
+import xbmcgui
+
+from simple_logging import SimpleLogging
+from resources.lib.functions import show_menu
+
+log = SimpleLogging(__name__)
+
+
+class ContextMonitor(threading.Thread):
+
+    stop_thread = False
+
+    def run(self):
+
+        item_id = None
+        log.debug("ContextMonitor Thread Started")
+        context_up = False
+        is_embycon_item = False
+
+        while not xbmc.abortRequested and not self.stop_thread:
```

Comment:
`xbmc.Monitor().abortRequested()` please

Reply:
This was picked up in the initial submission as well, in the other service loop I have, I did try to switch over at that time but found a bug in Kodi
Notes from my last attempt to use the monitor.

# monitor.abortRequested() is causes issues, it currently triggers for all addon cancelations which causes
# the service to exit when a user cancels an addon load action. This is a bug in Kodi.
# I am switching back to xbmc.abortRequested approach until kodi is fixed or I find a work around

Here is the forum thread on it
https://forum.kodi.tv/showthread.php?tid=319161&highlight=abortRequested

I am not sure this was pack ported to Kodi 17 yet however to maintain backward compatibility with all version of 17.x of Kodi I think I should stick with my current implementation for now and switch to the new one in Kodi 18.

Is that ok?

Reply:
@enen92 
Just wondering if this was ok.
Having to support multiple version of Kodi 17 with the same code base is my priority now.
I am not even sure how I would support addons for different versions of Kodi 17.x

Reply:
Oh ok I am using v18 for a while and I was not aware of this issue in Krypton. Please make sure you adapt if you add this addon to the Leia repo at some point. Thanks